rate-limiter/src/token_bucket.rs

```diff
@@ -44,12 +105,16 @@ impl TokenBucket {
 
     fn update_units(&mut self, now: Instant) -> usize {
         let time_since_last_update = now.duration_since(self.last_update);
-        let new_units = time_since_last_update
+        let mut new_units = time_since_last_update
             .as_micros()
             .saturating_mul(self.rate_per_second as u128)
             .saturating_div(1_000_000)
             .try_into()
             .unwrap_or(usize::MAX);
+        let used_new_units = min(new_units, self.requested);
```

Comment:
Out of curiosity: was it computed incorrectly previously?

Reply:
check the new version, please

Reply:
Much cleaner indeed